apps/src/lib/ui/accounts/AddParentEmailModal.jsx

```diff
@@ -0,0 +1,224 @@
+      this.setState({saveState: STATE_UNKNOWN_ERROR});
+    }
+  };
+
+  isFormValid(errors) {
+    return Object.keys(errors).every(key => !errors[key]);
+  }
+
+  getValidationErrors() {
+    const {errors} = this.state;
+    return {
+      parentEmail: errors.parentEmail || this.getNewEmailValidationError(),
+      emailOptIn: errors.emailOptIn || this.getEmailOptInValidationError()
+    };
+  }
+
+  getNewEmailValidationError = () => {
+    const {parentEmail} = this.state.values;
+    if (parentEmail.trim().length === 0) {
+      return i18n.addParentEmailModal_parentEmail_isRequired();
+    }
+    if (!isEmail(parentEmail.trim())) {
+      return i18n.addParentEmailModal_parentEmail_invalid();
+    }
+    return null;
+  };
+
+  getEmailOptInValidationError = () => {
+    const {emailOptIn} = this.state.values;
+    if (emailOptIn.length === 0) {
+      return i18n.addParentEmailModal_emailOptIn_isRequired();
+    }
+    return null;
+  };
+
+  onParentEmailChange = event => {
+    const {values, errors} = this.state;
+    values['parentEmail'] = event.target.value;
+    errors['parentEmail'] = undefined;
+    this.setState({values, errors});
+  };
+
+  onEmailOptInChange = event => {
+    const {values, errors} = this.state;
+    values['emailOptIn'] = event.target.value;
+    errors['emailOptIn'] = undefined;
+    this.setState({values, errors});
+  };
+
+  render = () => {
+    const {saveState, values} = this.state;
+    const validationErrors = this.getValidationErrors();
+    const isFormValid = this.isFormValid(validationErrors);
+    const saving = saveState === STATE_SAVING;
+    return (
+      <BaseDialog
+        useUpdatedStyles
+        isOpen
+        handleClose={this.cancel}
+        uncloseable={STATE_SAVING === saveState}
+      >
+        <div style={styles.container}>
+          <Header text={i18n.addParentEmailModal_title()} />
+          <Field
+            label={i18n.addParentEmailModal_parentEmail_label()}
+            error={validationErrors.parentEmail}
+          >
+            <input
+              type="email"
+              value={values.parentEmail}
+              disabled={saving}
+              tabIndex="1"
+              onKeyDown={this.onKeyDown}
+              onChange={this.onParentEmailChange}
+              autoComplete="off"
+              maxLength="255"
+              size="255"
+              style={styles.input}
+              ref={el => (this.parentEmailInput = el)}
+            />
+          </Field>
+          <Field error={validationErrors.emailOptIn}>
+            <div style={styles.emailOptIn}>
+              <label style={styles.label}>
+                {i18n.addParentEmailModal_emailOptIn_description()}{' '}
+                <a href={pegasus('/privacy')}>
+                  {i18n.changeEmailModal_emailOptIn_privacyPolicy()}
+                </a>
+              </label>
+              <div>
+                <div style={styles.radioButton}>
+                  <label style={styles.input}>
```

Comment:
why do we have labels without text, and why are the labels with the text not associated with their respective inputs?

Reply:
The labels without text are removed.

I'm not sure what you mean by the second bit.

Reply:
Labels can be explicitly associated with the inputs they are labeling to support screen readers and provide some interfacing assistance (clicking the label focuses the input, etc).

You can either wrap the input in the label, like

```html
<label>
  <input />
  {i18n.yes()}
</label>
```

or give the input an `id` and the label a matching `for` attribute, like

```html
<input id="yes" />
<label for="yes">{i18n.yes()}</label>
```

See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label for more info

Reply:
Thanks! I used the `htmlFor` attribute (React's version of `for`) for the labels